python-packages/securesync/devices/models.py

```diff
@@ -207,10 +207,15 @@ def set_key(self, key):
         self.key = key
 
     def get_key(self):
+        self.key = None
         if not self.key:
-            if self.get_metadata().is_own_device:
-                self.key = crypto.get_own_key()
-            elif self.public_key:
+            try:
+                if self.get_metadata().is_own_device:
+                    self.key = crypto.get_own_key()
+            except:
```

Comment:
Yeah, I was looking into that. It's a funky custom MySQL-specific error, so it's a bit messy to catch specifically.